tools/testbuild.sh

```diff
@@ -450,8 +450,8 @@ function refresh_cmake {
     kconfig-tweak --file $nuttx/build/.config -d $toolchain
   fi
 
-  if ! cmake --build build -t savedefconfig 1>/dev/null; then
-    cmake --build build -t savedefconfig
+  if ! cmake --build build -t nuttx_job_refresh 1>/dev/null; then
```

Comment:
why not use savedefconfig or migrate the action in nuttx_job_refresh to savedefconfig?

Reply:
Hi @xiaoxiang781216 
I created a simple test workflow with a NuttX repository with this PR by modifying the refresh_cmake function (testbuild.sh file) to cmake --build build -t savedefconfig.
There are these jobs sim-02 and arm-10 for Linux and macOS (even on macOS it must be possible to build NuttX with Cmake + Ninja)

https://github.com/simbit18/nuttx-testing-ci/actions/runs/11178303686/job/31075667531


linux
```
====================================================================================
Cmake in present: stm32f3discovery/usbnsh,CONFIG_ARM_TOOLCHAIN_CLANG
Configuration/Tool: stm32f3discovery/usbnsh,CONFIG_ARM_TOOLCHAIN_CLANG
2024-10-04 10:41:38
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Select HOST_LINUX=y
  Disabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Enabling CONFIG_ARM_TOOLCHAIN_CLANG
  Building NuttX...
On branch simbit-savedefconfig
Your branch is up to date with 'origin/simbit-savedefconfig'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   boards/arm/stm32/stm32f3discovery/configs/usbnsh/defconfig

no changes added to commit (use "git add" and/or "git commit -a")
====================================================================================
```

macOS

```
====================================================================================
Cmake in present: stm32_tiny/nsh,CONFIG_ARM_TOOLCHAIN_CLANG
Configuration/Tool: stm32_tiny/nsh,CONFIG_ARM_TOOLCHAIN_CLANG
2024-10-04 10:34:54
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Select HOST_MACOS=y
  Disabling CONFIG_ARM_TOOLCHAIN_BUILDROOT
  Enabling CONFIG_ARM_TOOLCHAIN_CLANG
  Building NuttX...
On branch simbit-savedefconfig
Your branch is up to date with 'origin/simbit-savedefconfig'.

Changes not staged for commit:
	modified:   boards/arm/stm32/stm32_tiny/configs/nsh/defconfig

no changes added to commit
====================================================================================
```

nuttx_job_refresh behaves with the same logic as the tools/refresh.sh script used for the refresh stage (make build) on GitHub.

See https://github.com/apache/nuttx/blob/master/tools/refresh.sh

```
example nucleo-l152re/nsh

./tools/refresh.sh --silent nucleo-l152re/nsh
+ rm -f SAVEconfig
+ mv .config SAVEconfig
+ cp -a boards/arm/stm32/nucleo-l152re/configs/nsh/defconfig .config
+ make olddefconfig
+ make savedefconfig
+ diff boards/arm/stm32/nucleo-l152re/configs/nsh/defconfig defconfig
```

Reply:
so, we should replace savedefconfig cmake function with your implementation, instead adding a new nuttx_job_refresh.

Reply:
@xiaoxiang781216  done